src/typescript/fable-splitter/index.ts

```diff
@@ -0,0 +1,172 @@
+    }
+}
+
+function ensureArray(obj: any) {
+    return (Array.isArray(obj) ? obj : obj != null ? [obj] : []);
+}
+
+// function getCommonBaseDir(filePaths: string[]) {
+//     function getCommonPrefix(parts: string[][]): string[] {
+//         function f(prefix: string[], xs: string[][]): string[] {
+//             if (xs.length === 0) {
+//                 return prefix;
+//             } else {
+//                 const x = xs[0];
+//                 let i = 0;
+//                 while (i < prefix.length && i < x.length && x[i] === prefix[i]) {
+//                     i = i + 1;
+//                 }
+//                 return f(prefix.slice(0, i), xs.slice(1));
+//             }
+//         }
+//         return parts.length === 0 ? [] : f(parts[0], parts.slice(1));
+//     }
+//     const normalized = filePaths.map((filePath) => Path.dirname(filePath).replace(/\\/g, "/").split("/"));
+//     return getCommonPrefix(normalized).join("/");
+// }
+
+function compileAST(babelAst: string, babelOptions?: babel.TransformOptions) {
+    const data = JSON.parse(babelAst);
+    if (data.error) { throw new Error(data.error); }
+
+    Object.keys(data.logs || {}).forEach((key) => {
+        const out = output(key);
+        ensureArray(data.logs[key]).forEach((log: string) => out(`${key} ==> ${log}`));
+    });
+
+    const code: string | undefined = undefined;
+    if (babelOptions && babelOptions.sourceMaps) {
+        babelOptions.sourceFileName = Path.relative(Process.cwd(), data.fileName.replace(/\\/g, "/"));
+    }
+    const result = Babel.transformFromAst(data, code, babelOptions);
+    return result;
+}
+
+async function transformAsync(path: string, options: FableCompilerOptions) {
+    // if already compiled, do nothing
+    let fullPath = Path.resolve(path).replace(/\\/g, "/");
+    if (alreadyCompiled.has(fullPath)) {
+        return undefined;
+    }
+    alreadyCompiled.add(fullPath);
+
+    let parsed: Babel.BabelFileResult | undefined;
+
+    // compile F#
+    if (FSHARP_EXT.test(path)) {
+        const fableMsg = JSON.stringify(Object.assign({}, options.fable, { path }));
+        const babelAst = await client.send(options.port, fableMsg);
+        parsed = compileAST(babelAst, options.babel);
+    } else {
+        // compile JS
+        fullPath = JAVASCRIPT_EXT.test(fullPath) ? fullPath : fullPath + ".js";
+        if (fs.existsSync(fullPath)) {
+            parsed = Babel.transformFileSync(fullPath, options.babel);
+        }
+    }
+
+    if (parsed) {
+        console.log(`Fable compiled: ${fullPath}`);
+
+        // if not a .fsproj, save as JS file
+        if (!FSPROJ_EXT.test(path)) {
+            // TODO: create common folder structure
+            const fileName = Path.basename(fullPath).replace(FSHARP_EXT, ".js");
+            const filePath = Path.resolve(Path.join(options.outDir, fileName));
+            const fileDir = Path.dirname(filePath);
+            if (!fs.existsSync(fileDir)) { fs.mkdirSync(fileDir); }
```

Comment:
I think this will fail if the parent dir doesn't exist either. [This function](https://github.com/fable-compiler/Fable/blob/6c8b94e5e1b7e30669a96646859f88f24376d934/src/fable/Fable.Client.Node/ts/lib.ts#L270-L281) should fix the problem.